.github/workflows/wasm.yml

```diff
@@ -0,0 +1,106 @@
+name: WebAssembly
+on:
+  push:
+    paths-ignore:
+      - 'doc/**'
+      - '**.md'
+      - '**.rdoc'
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+      - '**.md'
+      - '**.rdoc'
+
+concurrency:
+  group: ${{ github.workflow }} / ${{ startsWith(github.event_name, 'pull') && github.ref_name || github.sha }}
+  cancel-in-progress: ${{ startsWith(github.event_name, 'pull') }}
+
+jobs:
+  make:
+    strategy:
+      matrix:
+        entry:
+#         # wasmtime can't compile non-optimized Asyncified binary due to locals explosion
+#         - { name: O0-debuginfo, optflags: "-O0", debugflags: "-g", wasmoptflags: "-O1" }
+#         - { name: O1,           optflags: "-O1", debugflags: ""  , wasmoptflags: "-O1" }
+          - { name: O2,           optflags: "-O2", debugflags: ""  , wasmoptflags: "-O2" }
+#         - { name: O3,           optflags: "-O3", debugflags: ""  , wasmoptflags: "-O3" }
+#         # -O4 is equivalent to -O3 in clang, but it's different in wasm-opt
+#         - { name: O4,           optflags: "-O3", debugflags: ""  , wasmoptflags: "-O4" }
+#         - { name: Oz,           optflags: "-Oz", debugflags: ""  , wasmoptflags: "-Oz" }
```

Comment:
Wasmer should work great here (it shouldn't have the locals explosion when using the singlepass compiler)

It would be great to test it with wasmer here!

Reply:
Thank you for your info. But it looks wasmer also can't compile `-O0` binary in a reasonable time somehow. At least for now, I don't plan to test O0 on CI, sorry.

Reply:
Weird. Did you tried with `wasmer run ruby.wasm --singlepass`?
That should compile in linear time

Reply:
I'll file an issue for this to wasmer after this patch will be merged.